### ISSUE TITLE
tk toolmanager add_toolitem fails to add tool to group other than the last one

### ISSUE THREAD

**Issue by anntzer** [matplotlib/matplotlib#20130]
### Bug report

**Bug summary**

The list of default tools in backend_tools is currently defined as
```python
default_toolbar_tools = [['navigation', ['home', 'back', 'forward']],
                         ['zoompan', ['pan', 'zoom', 'subplots']],
                         ['io', ['save', 'help']]]
```
Let's just break the "zoompan" group in two:
```python
default_toolbar_tools = [['navigation', ['home', 'back', 'forward']],
                         ['zoompan', ['pan', 'subplots']],
                         ['io', ['save', 'help']],
                         ['zoompan', ['zoom']]]
```
(a more realistic case would be a third-party tool which wants to insert itself into a preexisting group).
For Qt, GTK3 and wx (open a figure while `rcParams["toolbar"] = "toolmanager"`), the "zoom" tool ends up in the same group (i.e., next to) "pan" and "subplots" (well, the ordering is slightly messed up, but let's think about that another time).  However, for tk, the "zoom" tool ends up by itself after all other tools.

Looking at the implementation of `add_toolitem`
```python
    def add_toolitem(
            self, name, group, position, image_file, description, toggle):
        frame = self._get_groupframe(group)
        button = NavigationToolbar2Tk._Button(self, name, image_file, toggle,
                                              lambda: self._button_click(name))
```
it seems likely that we should be doing something with `frame`...

**Code for reproduction**
**Actual outcome**
**Expected outcome**

discussed above.

**Matplotlib version**

  * Operating system: linux
  * Matplotlib version (`import matplotlib; print(matplotlib.__version__)`): HEAD/3.4.x
  * Matplotlib backend (`print(matplotlib.get_backend())`): tkagg
  * Python version: 39

**Comment by QuLogic** [comment]
I think this is broken in the `toolmanager_sgskip.py` example as well.

**Comment by anntzer** [comment]
Fixing the "appending" part is relatively easy:
```patch
diff --git i/lib/matplotlib/backends/_backend_tk.py w/lib/matplotlib/backends/_backend_tk.py
index 2f13408cd0..733ed8d477 100644
--- i/lib/matplotlib/backends/_backend_tk.py
+++ w/lib/matplotlib/backends/_backend_tk.py
@@ -822,7 +822,7 @@ class ToolbarTk(ToolContainerBase, tk.Frame):
     def add_toolitem(
             self, name, group, position, image_file, description, toggle):
         frame = self._get_groupframe(group)
-        button = NavigationToolbar2Tk._Button(self, name, image_file, toggle,
+        button = NavigationToolbar2Tk._Button(frame, name, image_file, toggle,
                                               lambda: self._button_click(name))
         if description is not None:
             ToolTip.createToolTip(button, description)
@@ -835,6 +835,7 @@ class ToolbarTk(ToolContainerBase, tk.Frame):
                 self._add_separator()
             frame = tk.Frame(master=self, borderwidth=0)
             frame.pack(side=tk.LEFT, fill=tk.Y)
+            frame._label_font = self._label_font
             self._groups[group] = frame
         return self._groups[group]
```
but this still fails to support the more general *position* arg which can insert tools at any position.

OTOH, one may ask whether we really need to support that, or whether we could live with a much simpler API which only supports taking a complete list of tools from the start...

**Comment by daniilS** [comment]
@anntzer I added the `position` arg to your fix in #22290
